### PR TITLE
Type of `to` in v2 message object

### PR DESCRIPTION
**Pull request by derelbenkoenig**

### CODE REVIEW

**Comment by dtolb** [comment]
Preview Changes at:
        http://bw-apdocs-v2-group-messages.s3-website-us-east-1.amazonaws.com/

File: messaging-2/methods/createGroup.md [hunk 1]
```diff
@@ -9,14 +9,14 @@ Send a group text message or picture message
...
-| applicationId | `string` | The ID of the Application your `from` number is associated with in the Bandwidth Phone Number Dashboard.     | Yes       |
-| media         | `array`  | A list of URLs to include as media attachments as part of the message.                                       | No        |
-| tag           | `string` | Any string which will be included in the callback events of the message.                                     | No        |
+| Parameter     | Type                             | Description                                                                                                  | Mandatory |
+|:--------------|:---------------------------------|:-------------------------------------------------------------------------------------------------------------|:----------|
+| from          | `string`                         | One of your telephone numbers the message should come from (must be in E.164 format, like +19195551212).     | Yes       |
+| to            | `string` or `array` of `string`  | The phone number or numbers the message should be sent to (must be in E.164 format, like +19195551212).      | Yes       |
```
Comment (dtolb) [inline messaging-2/methods/createGroup.md]: Is there anytime a group message **Won't be** an `array` of `strings`?

**Comment by dtolb** [comment]
I have some general questions about the `to` updated fields. I want to try to keep the single (normal) text message separate from the group message threads.

File: messaging-2/methods/createSingle.md [hunk 2]
```diff
@@ -9,14 +9,14 @@ Send a text message or picture message.
...
-| applicationId | `string` | The ID of the Application your `from` number is associated with in the Bandwidth Phone Number Dashboard.     | Yes       |
-| media         | `array`  | A list of URLs to include as media attachments as part of the message.                                       | No        |
-| tag           | `string` | Any string which will be included in the callback events of the message.                                     | No        |
+| Parameter     | Type                             | Description                                                                                                  | Mandatory |
+|:--------------|:---------------------------------|:-------------------------------------------------------------------------------------------------------------|:----------|
+| from          | `string`                         | One of your telephone numbers the message should come from (must be in E.164 format, like +19195551212).     | Yes       |
+| to            | `string` or `array` of `string`  | The phone number or numbers the message should be sent to (must be in E.164 format, like +19195551212).      | Yes       |
```
[the remark this answers is not in the document]
Comment (derelbenkoenig) [reply messaging-2/methods/createSingle.md]: Yeah, the way the code actually works is that the `to` field is always interpreted as a `Set` of strings. If the request JSON has a single string
```json
{
...
  to: "+12223334444",
... 
}

File: messaging-2/methods/createGroup.md (hunk 1 above)
Comment (derelbenkoenig) [reply messaging-2/methods/createGroup.md]: no, a group message will always be an array

**Comment by dtolb** [comment]
Preview Changes at:
        http://bw-apdocs-v2-group-messages.s3-website-us-east-1.amazonaws.com/

**Comment by dtolb** [comment]
Preview Changes at:
        http://bw-apdocs-v2-group-messages.s3-website-us-east-1.amazonaws.com/

**Comment by dtolb** [comment]
Preview Changes at:
        http://bw-apdocs-v2-group-messages.s3-website-us-east-1.amazonaws.com/

**Comment by dtolb** [comment]
Preview Changes at:
        http://bw-apdocs-v2-group-messages.s3-website-us-east-1.amazonaws.com/

**Comment by dtolb** [comment]
Preview Changes at:
        http://bw-apdocs-v2-group-messages.s3-website-us-east-1.amazonaws.com/

**Comment by dtolb** [comment]
Preview Changes at:
        http://bw-apdocs-v2-group-messages.s3-website-us-east-1.amazonaws.com/

**Comment by dtolb** [comment]
Preview Changes at:
        http://bw-apdocs-v2-group-messages.s3-website-us-east-1.amazonaws.com/

**Comment by derelbenkoenig** [comment]
@dtolb I don't think we'll get any media files that are `.txt`; that will just be appended to the `text` field on the message itself

**Comment by derelbenkoenig** [comment]
@dtolb why is one of the pages called `incomingGroup` and one called `receiveSingle`? Shouldn't they either both be `receive...` or both `incoming...`

**Comment by dtolb** [comment]
Preview Changes at:
        http://bw-apdocs-v2-group-messages.s3-website-us-east-1.amazonaws.com/

**Comment by dtolb** [comment]
Preview Changes at:
        http://bw-apdocs-v2-group-messages.s3-website-us-east-1.amazonaws.com/

**Comment by dtolb** [comment]
@derelbenkoenig 

> why is one of the pages called incomingGroup and one called receiveSingle? Shouldn't they either both be receive... or both incoming...

don't know, but I can update

File: messaging-2/events/incomingGroup.md [hunk 3]
```diff
@@ -48,7 +48,46 @@ User-Agent: BandwidthAPI/v2
...
+        "+12345678902",
+        "+12345678903"
+      ],
+      "from": "+12345678901",
+      "text": "Hey, check this out!",
+      "applicationId": "93de2206-9669-4e07-948d-329f4b722ee2",
+      "media": [
+        "https://api.catapult.inetwork.com/v1/users/{userId}/media/14762070468292kw2fuqty55yp2b2/0/bw.png",
+        "https://api.catapult.inetwork.com/v1/users/{userId}/media/14762070468292kw2fuqty55yp2b2/1/bandwidth_logo.png",
+        "https://api.catapult.inetwork.com/v1/users/{userId}/media/14762070468292kw2fuqty55yp2b2/2/Bandwidth_Contact.txt"
```
Comment (derelbenkoenig) [inline messaging-2/events/incomingGroup.md]: The current v2 behavior is to never include attachments that are `.txt`. I suppose this could change though

Comment (derelbenkoenig) [reply messaging-2/events/incomingGroup.md]: (any 'text' parts that are part of an incoming message are appended to the `text` field)

**Comment by dtolb** [comment]
Preview Changes at:
        http://bw-apdocs-v2-group-messages.s3-website-us-east-1.amazonaws.com/

**Pull request merged**